api/src/context/java/io/grpc/Deadline.java

```diff
@@ -225,37 +227,27 @@ public String toString() {
   @Override
   public int compareTo(Deadline that) {
     checkTicker(that);
-    long diff = this.deadlineNanos - that.deadlineNanos;
-    if (diff < 0) {
-      return -1;
-    } else if (diff > 0) {
-      return 1;
-    }
-    return 0;
+    return Long.compare(this.deadlineNanos, that.deadlineNanos);
   }
 
   @Override
   public int hashCode() {
-    return Arrays.asList(this.ticker, this.deadlineNanos).hashCode();
+    return Objects.hash(this.ticker, this.deadlineNanos);
   }
 
   @Override
-  public boolean equals(final Object o) {
-    if (o == this) {
+  public boolean equals(final Object object) {
```

Comment:
`o` is more traditional for the argument name

Reply:
Actually, the most common (according to google3) is `Object obj` believe it or not!

Reply:
I don't really disagree, but I already nit'ed multiple of the changes already. Kurt was trying to match go/java-practices/equals#impl which uses "object". I don't think we should go around rewriting all the code, but there's plenty of improvements in here to accept it on the whole. (I hadn't realized Objects has been available to us since API level 19).

Reply:
@kluever, interesting. I wonder if that's because of IDEs suggesting the name as pulled from the base class. https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)

(Not that it really matters. All of them are acceptable.)

Reply:
Yep, my team just reached the _exact_ same conclusion. I'm going to update the Java Practices on equals to use `Object obj` :-)

Reply:
See cl/747962435 (internally :-)